### PR TITLE
[LLVM][DOCS] Add documentation for 'host' and 'Native' options in LLVM_TARGETS_TO_BUILD.

### PR DESCRIPTION
**Pull request by harrisonGPU**
From https://github.com/llvm/llvm-project/issues/111356

### CODE REVIEW

File: llvm/docs/CMake.rst [hunk 1]
```diff
@@ -832,9 +832,15 @@ enabled sub-projects. Nearly all of these variable names begin with
 **LLVM_TARGETS_TO_BUILD**:STRING
   Semicolon-separated list of targets to build, or *all* for building all
   targets. Case-sensitive. Defaults to *all*. Example:
-  ``-DLLVM_TARGETS_TO_BUILD="X86;PowerPC"``.
+  ``-DLLVM_TARGETS_TO_BUILD="X86;PowerPC;host"``.
   The full list, as of March 2023, is:
-  ``AArch64;AMDGPU;ARM;AVR;BPF;Hexagon;Lanai;LoongArch;Mips;MSP430;NVPTX;PowerPC;RISCV;Sparc;SystemZ;VE;WebAssembly;X86;XCore``
+  ``AArch64;AMDGPU;ARM;AVR;BPF;Hexagon;Lanai;LoongArch;Mips;MSP430;NVPTX;PowerPC;RISCV;Sparc;SystemZ;VE;WebAssembly;X86;XCore;host;Native``
```
Comment (nikic) [inline llvm/docs/CMake.rst:837]: Listing additional options is a good idea, but I don't think they should be included in this list of targets, or the example above (esp. as for the majority of people, X86 is also host).

File: llvm/docs/CMake.rst [hunk 2]
```diff
@@ -832,9 +832,15 @@ enabled sub-projects. Nearly all of these variable names begin with
...
+
+  **Additional Options**:
+  - **host**: Automatically detect and build the target corresponding to the host machine's architecture.
+  
+    For example, on an x86_64 machine, `host` will map to `X86`.
```
Comment (nikic) [inline llvm/docs/CMake.rst:842]:
```suggestion
    For example, on an x86_64 machine, ``host`` will map to  `X86 `.
```

File: llvm/docs/CMake.rst [hunk 3]
```diff
@@ -832,9 +832,15 @@ enabled sub-projects. Nearly all of these variable names begin with
...
+  - **Native**: Alias for `host`, used interchangeably to specify the host architecture.
```
Comment (nikic) [inline llvm/docs/CMake.rst:843]: Given that they are the same, can we make this more compact by dropping the "Additional Options" header and just describing "host/Native"?

**Comment by PerMildner** [comment]
Not mentioned in the issue, but the value `all` is apparently also one of the "symbolic" values, so perhaps that too should be explicitly mentioned.

File: llvm/docs/CMake.rst [hunk 4]
```diff
@@ -836,6 +836,12 @@ enabled sub-projects. Nearly all of these variable names begin with
   The full list, as of March 2023, is:
   ``AArch64;AMDGPU;ARM;AVR;BPF;Hexagon;Lanai;LoongArch;Mips;MSP430;NVPTX;PowerPC;RISCV;Sparc;SystemZ;VE;WebAssembly;X86;XCore``
 
+  You can also specify ``host`` or ``Native`` to automatically detect and
```
Comment (MaskRay) [inline llvm/docs/CMake.rst:839]: Just document the common `host` and omit `Native`?

**Review by MaskRay — APPROVED** [review]

File: llvm/docs/CMake.rst (hunk 4 above)
Comment (PerMildner) [reply llvm/docs/CMake.rst:839]: It is good to only _promote_ one canonical symbolic name, e.g. `host`. However, for someone reading the documentation for understanding some project found on the net, it is really important that _all_ accepted names are documented.

Comment (harrisonGPU) [reply llvm/docs/CMake.rst:839]: Okay, thanks.

**Comment by harrisonGPU** [comment]
@MaskRay , Could you please help me merge this PR? Now I have not merge access right. :）

**Comment by MaskRay** [comment]
Other reviewers have commented. We leave the PR open for a while

**Review by RKSimon — APPROVED** [review]
LGTM

File: llvm/docs/CMake.rst [hunk 5]
```diff
@@ -836,6 +836,12 @@ enabled sub-projects. Nearly all of these variable names begin with
...
+  include the target corresponding to the host machine's architecture, or
+  use ``all`` to include all available targets.
```
Comment (nikic) [inline llvm/docs/CMake.rst:841]: Note that all is already mentioned above (in the part that github helpfully hides by default).

File: llvm/docs/CMake.rst (hunk 3 above)
Comment (harrisonGPU) [reply llvm/docs/CMake.rst]: Okay, I have already removed it.

File: llvm/docs/CMake.rst (hunk 5 above)
Comment (harrisonGPU) [reply llvm/docs/CMake.rst:841]: Okay, thanks!

File: llvm/docs/CMake.rst (hunk 1 above)
Comment (harrisonGPU) [reply llvm/docs/CMake.rst]: Done.

**Comment by harrisonGPU** [comment]
Thanks！

**Comment by PerMildner** [comment]
Thanks all for the speedy fix!

**Pull request merged**